### DISCUSSION TITLE
How to compile the ODBC drivers?

### DISCUSSION THREAD

**Discussion by kkovacs** [duckdb/duckdb#15189] [General]
Hi fellow Ducks,

first of all, thanks for the incredible job you are doing with DuckDB!

_TLDR: I don't seem to be able to compile a functioning ODBC driver. Please advise._

I'd like to have a look into issue https://github.com/duckdb/duckdb/issues/11380 , _"ODBC Driver doesn't handle path to database correctly"_, because it's a blocking issue for us too.

I wanted to compile the ODBC drivers on Linux first, but turns out it's not as easy as I thought - I would need some pointers. Maybe the instructions are outdated, or maybe I'm just thick headed :)

(Also, the [ODBC downloads](https://duckdb.org/docs/installation/?version=stable&environment=odbc&platform=linux&download_method=direct&architecture=x86_64) are behind the "normal" DuckDB versions (1.1.0 vs 1.1.3), so maybe there is some problem momentarily?)

## First attempt

So, in the https://github.com/duckdb/duckdb-odbc repository, the `README.md` starts with:

> Build the ODBC client (from within the main DuckDB repository)
> `BUILD_ODBC=1 DISABLE_SANITIZER=1 make debug -j`

I assumed that the "main DuckDB repository" is the https://github.com/duckdb/duckdb one, but running the above command there does NOT produce a `libduckdb_odbc.so` file, which would be the ODBC driver.

```bash
$ find . -name '*.so'
./build/debug/tools/sqlite3_api_wrapper/libsqlite3_api_wrapper.so
./build/debug/src/libduckdb.so
```

No `libduckdb_odbc.so`.

## Second attempt

Apparently the https://github.com/duckdb/duckdb-odbc repo is regularly updated ("vendored") from the https://github.com/duckdb/duckdb repo (although strangely, the mentioned hashes are not found in the original repo):

```
ed948a5  32 hours ago DLGBot          o Update vendored DuckDB sources to 73b56c6
73b56c6    2 days ago DLGBot          o Update vendored DuckDB sources to 091af07
091af07    3 days ago DLGBot          o Update vendored DuckDB sources to 268b4b6
```

So I tried to run the following in the https://github.com/duckdb/duckdb-odbc repo next:

```bash
BUILD_ODBC=1 DISABLE_SANITIZER=1 make release -j 4
```

This does produce a `libduckdb_odbc.so`, but when I try to use it, I get the following error, indicating that somehow the compile is not "complete", it misses "core" features:

```bash
$ ./linux_setup/unixodbc_setup.sh -u -D ~/duckdb-odbc/build/release/libduckdb_odbc.so

$ echo "select version();" | isql -v DuckDB
+---------------------------------------+
| Connected!                            |
|                                       |
| sql-statement                         |
| help [tablename]                      |
| echo [string]                         |
| quit                                  |
|                                       |
+---------------------------------------+
SQL> select version();
[37000]ODBC_DuckDB->PrepareStmt
Catalog Error: Scalar Function with name "version" is not in the catalog, but it exists in the core_functions extension.

Please try installing and loading the core_functions extension by running:
INSTALL core_functions;
LOAD core_functions;

Alternatively, consider enabling auto-install and auto-load by running:
SET autoinstall_known_extensions=1;
SET autoload_known_extensions=1;
[ISQL]ERROR: Could not SQLPrepare
```

Basically nothing works, since "core_functions" does not seem to be compiled in.

The suggested INSTALL/LOAD does not work either (but I'm pretty sure this should not be needed anyway, core should probably be built in...):

```
SQL> INSTALL core_functions;
SQLRowCount returns -1
SQL> LOAD core_functions;
[S1000]ODBC_DuckDB->SingleExecuteStmt
IO Error: Extension "/home/user/.duckdb/extensions/aa3787b0c1/linux_amd64/core_functions.duckdb_extension" could not be loaded because its signature is either missing or invalid and unsigned extensions are disabled by configuration (allow_unsigned_extensions)
[ISQL]ERROR: Could not SQLExecute
```

Please advise! :)

_Edit: typo_

**Comment by kkovacs** [comment]
Closing here, moved to https://github.com/duckdb/duckdb/issues/15254.
